core/modules/system/system.install

```diff
@@ -672,6 +672,48 @@ function system_install() {
   // Populate the cron key variable.
   $cron_key = backdrop_random_key();
   state_set('cron_key', $cron_key);
+
+  // Set the site_mail to a sensible default.
+  $site_mail = 'no-reply@' . backdrop_get_bare_domain($_SERVER['SERVER_NAME']);
+  if (valid_email_address($site_mail)) {
+    config_set('system.core', 'site_mail', $site_mail);
+  }
+}
+
+/**
+ * Take a full URL and return only the bare domain.
+ *
+ * @param (string) $url
+ *   A fully-qualified URL like https://www.example.co.uk/path-to-page.
+ *
+ * @return (string)
+ *   The bare domain of the starting URL like example.com or example.co.uk.
+ */
+function backdrop_get_bare_domain($url) {
```

Comment:
We also need new testing to cover this function with various inputs such as with or without www, two-letter domain names and TLDs, multiple subdomains, etc.